ld-pruning-wf.wdl

```diff
@@ -0,0 +1,441 @@
+		f.write("variant_include_file " + variant_include_file + "\n")
+
+		if "~{defSampleInclude}" == "true":
+			f.write("sample_include_file ~{sample_include_file}\n")
+
+		if ("~{out_prefix}" != ""):
+			chromosome = py_rootPlusChr(gds)[1]
+			py_filename = "~{out_prefix}" + "_chr" + chromosome + ".gds"
+		else:
+			chromosome = py_rootPlusChr(gds)[1]
+			basename = os.path.basename(py_rootPlusChr(gds)[0])
+			py_filename = basename + "subset_chr" + chromosome + ".gds"
+
+		f.write("subset_gds_file " + py_filename)
+		f.close()
+		CODE
+
+		R -q --vanilla < /usr/local/analysis_pipeline/R/subset_gds.R --args subset_gds.config
+	}
+
+	runtime {
+		cpu: cpu
+		docker: "uwgac/topmed-master:2.10.0"
+		disks: "local-disk " + final_disk_dize + " HDD"
+		memory: "${memory} GB"
+		preemptibles: "${preempt}"
+	}
+	output {
+		File config_file = "subset_gds.config"
+		File subset_output = glob("*.gds")[0]
+	}
+}
+
+# [3] merge_gds -- merge a bunch of GDS files into one
+task merge_gds {
+	input {
+		Array[File] gdss
+		String? out_prefix
+
+		# runtime attributes
+		Int addldisk = 5
+		Int cpu = 2
+		Int memory = 4
+		Int preempt = 3
+	}
+
+	# Estimate disk size required
+	Int gds_size = ceil(size(gdss, "GB"))
+	Int final_disk_dize = gds_size * 3 + addldisk
+	String filename = select_first([out_prefix, "merged"])
+
+	command <<<
+		set -eux -o pipefail
+
+		# CWL has an ln -s, will probably need to use copy trick again
+		echo "Copying inputs into the workdir"
+		BASH_FILES=(~{sep=" " gdss})
+		for BASH_FILE in ${BASH_FILES[@]};
+		do
+			cp ${BASH_FILE} .
+		done
+
+		echo "Generating config file"
+		python << CODE
+		import os
+
+		def find_chromosome(file):
+			chr_array = []
+			chrom_num = split_on_chromosome(file)
+			if(unicode(str(chrom_num[1][1])).isnumeric()):
+				# two digit number
+				chr_array.append(chrom_num[1][0])
+				chr_array.append(chrom_num[1][1])
+			else:
+				# one digit number or Y/X/M
+				chr_array.append(chrom_num[1][0])
+			return "".join(chr_array)
+
+		def split_on_chromosome(file):
+			# if input is "amishchr1.gds"
+			# output is ["amish", ".gds", "chr"]
+			chrom_num = file
+			if "chr" in chrom_num:
+				chrom_num = chrom_num.split("chr")
+				chrom_num.append("chr")
+			else:
+				return "error-invalid-inputs"
+			return chrom_num
+
+		def write_config(chr_array, precisely_one_gds_split):
+			f = open("merge_gds.config", "a")
+			f.write("chromosomes ")
+			f.write("'")
+			for chr in chr_array:
+				f.write(chr)
+				f.write(" ")
+			f.write("'")
+			f.write("\ngds_file ")
+			f.write("'")
+			f.write(precisely_one_gds_split[0])  # first part
+			f.write(precisely_one_gds_split[2])  # string "chr"
+			f.write(" ")  # space where R script inserts chr number
+			if(unicode(str(precisely_one_gds_split[1][1])).isnumeric()):
+				# two digit number
+				f.write(precisely_one_gds_split[1][2:])
+			else:
+				# one digit number or Y/X/M
+				f.write(precisely_one_gds_split[1][1:])
+			f.write("'")
```

Comment:
This should be written on one line using the fact that you already know how to pull the chr. (So you don't need to rewrite that code)